modules/rostests/kmtests/ntos_io/IoDeviceInterface.c

```diff
@@ -1,125 +1,214 @@
+    0x760FED5EL, 0x9357, 0x11D0, {0xA3, 0xCC, 0x00, 0xA0, 0xC9, 0x22, 0x31, 0x96}
+#define STATIC_KSCATEGORY_INTERFACETRANSFORM \
+    0xCF1DDA2DL, 0x9743, 0x11D0, {0xA3, 0xEE, 0x00, 0xA0, 0xC9, 0x22, 0x31, 0x96}
+#define STATIC_KSCATEGORY_MEDIUMTRANSFORM \
+    0xCF1DDA2EL, 0x9743, 0x11D0, {0xA3, 0xEE, 0x00, 0xA0, 0xC9, 0x22, 0x31, 0x96}
+#define STATIC_KSCATEGORY_MIXER \
+    0xAD809C00L, 0x7B88, 0x11D0, {0xA5, 0xD6, 0x28, 0xDB, 0x04, 0xC1, 0x00, 0x00}
+#define STATIC_KSCATEGORY_RENDER \
+    0x65E8773EL, 0x8F56, 0x11D0, {0xA3, 0xB9, 0x00, 0xA0, 0xC9, 0x22, 0x31, 0x96}
+#define STATIC_KSCATEGORY_SPLITTER \
+    0x0A4252A0L, 0x7E70, 0x11D0, {0xA5, 0xD6, 0x28, 0xDB, 0x04, 0xC1, 0x00, 0x00}
+
+static const GUID KSCATEGORY_BRIDGE = { STATIC_KSCATEGORY_BRIDGE };
+static const GUID KSCATEGORY_CAPTURE = { STATIC_KSCATEGORY_CAPTURE };
+static const GUID KSCATEGORY_COMMUNICATIONSTRANSFORM = { STATIC_KSCATEGORY_COMMUNICATIONSTRANSFORM };
+static const GUID KSCATEGORY_DATACOMPRESSOR = { STATIC_KSCATEGORY_DATACOMPRESSOR };
+static const GUID KSCATEGORY_DATADECOMPRESSOR = { STATIC_KSCATEGORY_DATADECOMPRESSOR };
+static const GUID KSCATEGORY_DATATRANSFORM = { STATIC_KSCATEGORY_DATATRANSFORM };
+static const GUID KSCATEGORY_FILESYSTEM = { STATIC_KSCATEGORY_FILESYSTEM };
+static const GUID KSCATEGORY_INTERFACETRANSFORM = { STATIC_KSCATEGORY_INTERFACETRANSFORM };
+static const GUID KSCATEGORY_MEDIUMTRANSFORM = { STATIC_KSCATEGORY_MEDIUMTRANSFORM };
+static const GUID KSCATEGORY_MIXER = { STATIC_KSCATEGORY_MIXER };
+static const GUID KSCATEGORY_RENDER = { STATIC_KSCATEGORY_RENDER };
+static const GUID KSCATEGORY_SPLITTER = { STATIC_KSCATEGORY_SPLITTER };
+
+static const GUID* Types[] =
 {
-   NTSTATUS Status;
-   PWSTR SymbolicLinkList;
-   PWSTR SymbolicLinkListPtr;
-   GUID Guid = {0x378de44c, 0x56ef, 0x11d1, {0xbc, 0x8c, 0x00, 0xa0, 0xc9, 0x14, 0x05, 0xdd}};
-
-   Status = IoGetDeviceInterfaces(
-      &Guid,
-      NULL,
-      0,
-      &SymbolicLinkList);
-
-   ok(NT_SUCCESS(Status),
-         "IoGetDeviceInterfaces failed with status 0x%X\n",
-         (unsigned int)Status);
-   if (!NT_SUCCESS(Status))
-   {
-      return;
-   }
-
-   DPRINT("IoGetDeviceInterfaces results:\n");
-   for (SymbolicLinkListPtr = SymbolicLinkList;
-        SymbolicLinkListPtr[0] != 0 && SymbolicLinkListPtr[1] != 0;
-        SymbolicLinkListPtr += wcslen(SymbolicLinkListPtr) + 1)
-   {
-      DPRINT1("Symbolic Link: %S\n", SymbolicLinkListPtr);
-   }
+    &KSCATEGORY_BRIDGE,
+    &KSCATEGORY_CAPTURE,
+    &KSCATEGORY_COMMUNICATIONSTRANSFORM,
+    &KSCATEGORY_DATACOMPRESSOR,
+    &KSCATEGORY_DATADECOMPRESSOR,
+    &KSCATEGORY_DATATRANSFORM,
+    &KSCATEGORY_FILESYSTEM,
+    &KSCATEGORY_INTERFACETRANSFORM,
+    &KSCATEGORY_MEDIUMTRANSFORM,
+    &KSCATEGORY_MIXER,
+    &KSCATEGORY_RENDER,
+    &KSCATEGORY_SPLITTER,
+};
 
-#if 0
-   DPRINT("[PnP Test] Trying to get aliases\n");
+static
+VOID
+Test_IoOpenDeviceInterfaceRegistryKey(PWSTR SymbolicLink)
```

Comment:
Nit: Seems like you can use `PCWSTR` in all of these functions

Reply:
Done.